packages/components/src/pages/ChatSearch.vue

```diff
@@ -497,6 +503,47 @@ export default {
       const metadata = await this.rpcClient.metadata();
       this.metadata = metadata;
     },
+    async onDrop(evt: any) {
+      const items = evt.dataTransfer.items;
+      const len = items.length;
+      let promises: Promise<PinFileRequest[]>[] = [];
+      for (let i = 0; i < len; i++) {
+        const item = items[i];
+        if (item.kind === 'string' && item.type?.match('^application/vnd.code.uri-list')) {
+          promises.push(
+            new Promise((resolve) => {
+              item.getAsString((s: string) => {
+                resolve(
+                  s.split(/[\r\n]+/).map((uri) => {
+                    return {
+                      name: uri,
+                      uri,
+                    };
+                  })
+                );
+              });
+            })
+          );
+        } else if (item.kind === 'file') {
+          const file = item.getAsFile();
+          const name = 'file:///' + file.name;
+          promises.push(Promise.resolve([{ name, file }]));
+        }
+      }
+      await Promise.all(promises).then((requests) => {
```

Comment:
The `onDrop` method lacks error handling for asynchronous file reading and URI processing, which can lead to unhandled promise rejections.
\- Navie

DB: After looking through MDN, I didn't see any explicit mention of exceptions raised by the methods called, maybe it's not a consideration.